Python/optimizer_bytecodes.c

```diff
@@ -467,6 +467,15 @@ dummy_func(void) {
         res = sym_new_truthiness(ctx, value, false);
     }
 
+    op(_UNARY_NEGATIVE, (value -- res)) {
+        if (sym_matches_type(value, &PyLong_Type) || sym_matches_type(value, &PyFloat_Type)) {
+            res = sym_new_type(ctx, sym_get_type(value));
```

Comment:
This is a more efficient way to express the same idea (it avoids potentially looking up the type three times):

```suggestion
        PyTypeObject *type = sym_get_type(value);
        if (type == &PyLong_Type || type == &PyFloat_Type) {
            res = sym_new_type(ctx, type);
```

Reply:
Much better - thanks!